optimization/physical_constants.py

```diff
@@ -16,44 +16,88 @@ def fuel_cond(T):
-rho_W = 19250  # clad density [kg/m^3]
-rho_UN = 11300  # fuel density [kg/m^3]
-fuel_frac = 0.6  # fraction of fuel in CERMET
+const = {'T_center' : 1847.5, # centerline fuel temp (max) [K]
+         'k_clad' : 108.3,  # clad conductivity: W @ 1473 K [W/m-K]
+         'rho_W' : 19250,  # clad density [kg/m^3]
+         'rho_UN' : 11300,  # fuel density [kg/m^3]
+         'fuel_frac' : 0.6,  # volume fraction of fuel in CERMET
+         }
+# mixed fuel density
+const.update( {'rho_fuel' : const['fuel_frac'] * const['rho_UN'] +
+                           (1 - const['fuel_frac'])*const['rho_W']} )
+# conservative estimate (@centerline T) for fuel thermal conductivity
+const.update( {'k_fuel' : fuel_cond(const['T_center'])} )
 
-###############################################################################
-#                                                                             #
-#                   Calculated Parameters From Power Cycle                    #
-#                                                                             #
-###############################################################################
-dp_allowed = P_in - P_out  # pressure drop limit [Pa]
-Pr = Cp_cool * mu / k_cool  # coolant Prandtl numbe[-]
-# mixed density for CERMET fuel
-rho_fuel = fuel_frac*rho_UN + (1-fuel_frac)*rho_W
+class FlowProperties:
+    """Class to store flow properties and calculate secondary properties from
+    fundamental props.
+    """
+    # default flow properties
+    defaults = {'m_dot' : 0.75, # coolant flow [kg/s]
+                'Q_therm' : 131000, # core thermal power [W]
+                'T' : 1031.45, # bulk coolant temp [K]
+                'P' : 1.766e7, # bulk coolant pressure [Pa]
+                'dp_limit' : 483500, # pressure drop limit [Pa]
+               }
+
+    def __init__(self, **kwargs):
+        """Inialize FlowProperties class and load required flow property data.
+
+        Modified Attributes:
+        -----------------------
+            m_dot: (float) mass flow rate [kg/s]
+            Q_therm: (float) required thermal reactor power [W]
+            T: (float) bulk coolant temperature [K]
+            P: (float) bulk coolant pressure [Pa]
+            dp_limit: (float) power-cycle constrained dp [Pa]
+        """
+        if kwargs:
+            self.m_dot = kwargs['m_dot']
+            self.Q_therm = kwargs['Q_therm']
+            self.T = kwargs['T']
+            self.P = kwargs['P']
+            self.dp_limit = kwargs['dp_limit']
+        else: # use default values
+            for property in self.defaults:
+                self.__dict__[property] = self.defaults[property]
+            print("Warning, default flow properties are for testing purposes!!")
+
+        # estimate secondary properties 
+        self.secondary_properties()
+        # if the input temperature is out of range of the fit, print a warning
+        # message
+        if self.T < self.t_limit[0] or self.T > self.t_limit[1]:
```

Comment:
Not sure we need to set these limits as member variable - issue this warning below where `t_limit` is defined and define it as a local variable.